src/inputs/labeled-inputs.scss

```diff
@@ -101,20 +112,30 @@
     > .iui-message {
       margin: 0;
 
-      svg {
+      > svg,
+      :where(:not(.iui-button)) svg {
         margin-right: 0;
       }
     }
 
     > .iui-input,
     > .iui-textarea {
-      padding-right: $iui-xl;
+      padding-right: $iui-icons-default + $iui-l;
 
       + .iui-message {
         position: absolute;
         top: 50%;
-        right: $iui-sm;
+        right: 0;
         transform: translateY(-50%);
+
+        > svg,
+        :where(:not(.iui-button)) svg {
+          margin-right: $iui-sm;
+        }
+      }
+
+      &:last-child {
```

Comment:
I guess you could use nth-child selector for non inline input to achieve same, if you wanted...

Reply:
This is what @FlyersPh9 said to me:

> the borderless button is within the message class.
> 
> ```
> <label/>
> <input/>
> <message>
>   <button>
> </message>
> ```
> 
> so i cant think of a way to apply padding to the input based on if a button is present within message. 🤔

I don't see how `nth-child` would help with that. The html structure is the same.

I think this is why we need to avoid `absolute` positioning, and try to use flex instead. But that's for future I guess.

Reply:
Oh yeah, now I see.. the html is just wrong.. We cannot have message container, where part of it is under input, part of it is in input.. It is just wrong, where elements are magically appearing on totally different places. 
We might need to refactor this thing... again... Cause we already saw some use cases, where this format just does not work. We might just want separate sections tackling with css:
label above input
input
right input icon
message below.